File: 49-GroupAnagrams/49-GroupAnagrams.py
# Last updated: 3/14/2026, 4:49:58 PM
class Solution:
        def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
            anagram_dict = {}
            outer_list = []
            for word in strs:
                temp_str = ''.join(sorted(word))
                anagram_dict[temp_str] = anagram_dict.get(temp_str,[])
                anagram_dict[temp_str].append(word)
            for value in anagram_dict.values():
                outer_list.append(value)
            return outer_list

File: 49-GroupAnagrams/49-GroupAnagrams.java
// Last updated: 3/14/2026, 11:51:08 AM
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();

        for(int i = 0 ; i < strs.length; i++){
            char[] array = strs[i].toCharArray();
            Arrays.sort(array);
            String temp = String.valueOf(array);

            if(map.containsKey(temp)){
                map.get(temp).add(strs[i]);
            }
            else{
                List<String> list = new ArrayList<>();
                list.add(strs[i]);
                map.put(temp,list);
            }
        }
        List<List<String>> ll = new ArrayList<>(map.values());
        return ll;
    }
}